include/libtorrent/alloca.hpp

```diff
@@ -33,21 +33,28 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifndef TORRENT_ALLOCA
 
 #include "libtorrent/config.hpp"
+#include "libtorrent/span.hpp"
 
 #if defined TORRENT_WINDOWS || defined TORRENT_MINGW
 
 #include <malloc.h>
-#define TORRENT_ALLOCA(t, n) static_cast<t*>(_alloca(sizeof(t) * (n)))
+#define TORRENT_ALLOCA(v, t, n) ::libtorrent::span<t> v; { \
```

Comment:
oh, you specify the name of the local variable you want the span to be assigned to.. I see. because if you ever misuse the macro as: ``auto mem = TORRENT_ALLOCA()`` you'll always end up with an empty span.